tests/test_agents.py

```
@@ -760,6 +778,44 @@ def test_provide_final_answer(self, images, expected_messages_list):
                 for content, expected_content in zip(message["content"], expected_message["content"]):
                     assert content == expected_content
 
+    @pytest.mark.parametrize(
+        "tools, managed_agents, expectation",
+        [
+            # Valid case: no duplicates
+            (
+                [MockTool("tool1"), MockTool("tool2")],
+                [MockAgent("agent1", [MockTool("tool3")])],
+                does_not_raise(),
+            ),
+            # Invalid case: duplicate tool names
+            ([MockTool("tool1"), MockTool("tool1")], [], pytest.raises(ValueError)),
+            # Invalid case: tool name same as managed agent name
+            ([MockTool("tool1")], [MockAgent("tool1", [MockTool("final_answer")])], pytest.raises(ValueError)),
+            # Invalid case: tool name same as managed agent's tool name
+            ([MockTool("tool1")], [MockAgent("agent1", [MockTool("tool1")])], pytest.raises(ValueError)),
+            # Invalid case: duplicate managed agent name and managed agent tool name
+            ([], [MockAgent("tool1", [MockTool("tool1")])], pytest.raises(ValueError)),
```

Same as above, I'm not sure that this should raise an error.

---

Done.